pcsx2/x86/ix86-32/iR5900-32.cpp

```diff
@@ -1373,6 +1373,25 @@ void recompileNextInstruction(int delayslot)
 
 	cpuRegs.code = *(int *)s_pCode;
 
+	// Hardcoded Ratchet & Clank fixes since the game dyamically addresses memory for the program
+	// meaning it cannot be patched by traditional methods.
+	if (!EmuConfig.Gamefixes.RatchetDynaHack)
+	{
+		if (cpuRegs.code == 0x4B8123BC)
+		{
+			if (*(int*)PSM(pc + 0x4) == 0x4BE30858 && *(int*)PSM(pc + 0x8) == 0xF8610010 && *(int*)PSM(pc + 0xc) == 0x4A20009C)
+			{
```

Comment:
DevCon.Warning("Ratchet patch applied at %x", pc);

It's best to keep track of what the patch does even on retail builds and this doesn't introduce a performance hit too.

Reply:
eh, I could, I don't really want a bunch of stuff showing in retail builds.